src/osyris/core/array.py

```diff
@@ -253,6 +256,24 @@ def __mul__(self, other):
                                   unit=1.0 * result.units)
         return self.__class__(values=self._array * other, unit=self._unit)
 
+    def __matmul__(self, other):
+        if isinstance(other, self.__class__):
+            scale_l = self._unit.to_base_units()
+            scale_r = other._unit.to_base_units()
+            result = scale_l * scale_r
+            lhs = self._array
+            rhs = other._array * result.magnitude
+            lhs, rhs = self._broadcast(lhs, rhs)
+            return self.__class__(values=np.transpose(lhs @ rhs),
+                                  unit=1.0 * result.units)
+        if isinstance(other, Quantity):
```

Comment:
Does it even make sense to have the case with a single number (with unit)? You are doing matrix multiplication, and a `Quantity` cannot be a matrix

Reply:
My mistake. Forgot to remove that bit of code.